certificate-rotation.md

```diff
@@ -0,0 +1,113 @@
+
+## Generating new CA certificates
+
+CA rotation includes generating a new CA certificate and key.
+
+If the original certificate and key are generated using:
+```
+$ cockroach cert create-ca --certs-dir=certs --ca-key=ca.key --logtostderr=true
+I170425 14:20:29.394473 1 security/certs.go:120  Generated CA key ca.key
+I170425 14:20:29.398081 1 security/certs.go:173  Wrote 1 certificates to certs/ca.crt
+```
+
+We can generate a new CA certificate and key with the following:
+```
+$ cockroach cert create-ca --certs-dir=certs --ca-key=ca.new.key --overwrite --logtostderr=true
+I170425 14:21:10.490341 1 security/certs.go:120  Generated CA key ca.new.key
+I170425 14:21:10.494007 1 security/certs.go:160  Found 1 certificates in certs/ca.crt
+I170425 14:21:10.494341 1 security/certs.go:173  Wrote 2 certificates to certs/ca.crt
+```
+
+Flags:
+* `--ca-key=ca.new.key`: path to the new key to generate. To re-use the existing key (not recommended), use `--ca-key=ca.key` and `--allow-key-reuse`.
+* `--overwrite`: add the new certificate to the existing `ca.crt` file.
+
+This results in a `ca.crt` containing the new certificate followed by the old certificate. This
+combined CA certificate should be used by all nodes and clients to verify old node/client certificates
+and well as new certificates.
+
+## Generating new node/client certificates
+
+Client and node certificates are simpler than CA certificates because we do not need to keep old
+certificates around.
+
+Generating a new certificate and key is the same as the original generation. If the files already exist, specify `--overwrite` to overwrite both certificate and key.
+
+If the original certificate and key for node `localhost` were created with:
+```
+$ cockroach cert create-node --certs-dir=certs --ca-key=ca.key --logtostderr=true localhost
+I170425 14:28:10.713448 1 security/certs.go:226  Generated node certificate: certs/node.crt
+I170425 14:28:10.713845 1 security/certs.go:232  Generated node key: certs/node.key
+```
+
+We can either delete the `node.crt` and `node.key` files and rerun the same command, or run:
+```
+roach cert create-node --certs-dir=certs --ca-key=ca.key --logtostderr=true --overwrite localhost
+I170425 14:28:45.839331 1 security/certs.go:226  Generated node certificate: certs/node.crt
+I170425 14:28:45.839556 1 security/certs.go:232  Generated node key: certs/node.key
+```
+
+## Deploying certificates
+
+Cockroach does not currently provide any tools to deploy certificates and keys to clients and nodes.
+The appropriate files should be copied to the hosts needing them. Specifically:
+
+* each node needs: `ca.crt`, `node.crt`, `node.key`, with the node certificate issued for its hostnames and IPs
+* a client with username `marc` needs: `ca.crt`, `client.marc.crt`, `client.marc.key`.
+
+When generated a new CA certificate and adding it to the existing file, you **must** push the newly 
```

Comment:
This requirement is because the exiting node/client certs depend on the old CA cert, right? We should probably spell this out more.